File: go_solutions/solutions/int_to_roman.go
package solutions

func intToRoman(num int) string {
	//TODO - Started in Go and moved to Python
	result := ""
	romanDigits := map[int]string{
		1:    "I",
		5:    "V",
		10:   "X",
		50:   "L",
		100:  "C",
		500:  "D",
		1000: "M",
	}
	for num > 0 {
		msDigit := getMSDigit(num)
		if msDigit == 4 {
			result += romanDigits[1] + romanDigits[5]
		} else if msDigit == 9 {
			result += romanDigits[1] + romanDigits[10]
		} else {
			result += romanDigits[msDigit]
		}

	}
	return ""
}

func getMSDigit(num int) int {
	factor := 1
	for num > 10 {
		num = num / 10
	}
	return num * (10 ^ factor)
}

File: py_soltions/int_to_roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        all_integers = [1,5,10,50,100,500,1000,4,9,40,90,400,900]
        all_romans = ["I","V","X","L","C","D","M","IV","IX","XL","XC","CD","CM"]
        integers = [1,5,10,50,100,500,1000]
        roman = ["I","V","X","L","C","D","M"]
        res = ""
        while num > 0:
            ms_digit = get_ms_digit(num)
            if ms_digit in all_integers:
                num -= ms_digit
                res += all_romans[all_integers.index(ms_digit)]
            else:
                for i in range(len(integers) - 1, -1, -1):
                    if ms_digit > integers[i]:
                        num -= integers[i]
                        res += roman[i]
                        break
        return res


def get_ms_digit(num):
    factor = 0
    while num > 10:
        num = int(num / 10)
        factor += 1
    return num * (10 ** factor)

if __name__ == '__main__':
    s = Solution()
    print(s.intToRoman(3749))
    assert s.intToRoman(3749) == "MMMDCCXLIX"